Machine_Learning/README.md

```diff
@@ -624,7 +634,31 @@ All of the things above are "unwanted" but it is important to understand that me
+
+
+### Batch size:
+
+The "batch size" refers to the number of samples (image patches) that are processed by the neural network at once. During each epoch, the training dataset is typically divided into several smaller batches, and the neural network is updated based on the average loss across the batch. The batch size is another user-defined parameter that controls how many samples are processed in each batch. If the batch size is too small, the model may converge slowly and can result in poor generalization. If the batch size is too large, it may result in the model being unable to fit into memory, slowing down the training process, and potentially leading to poor generalization. The default batch size in ZEN is 64, but it is advisable to experiment with smaller sizes, such as 32 or 16, if memory errors occur.
+
+### Window size:
+
+The "window size" parameter determines the size of the image patch that N2V uses to make predictions. Specifically, N2V divides the input image into overlapping patches of a fixed size (determined by the window size parameter) and trains a neural network to predict the central pixel of each patch based on the other pixels in the patch. During inference (denoising), N2V processes the input image in a sliding-window fashion, applying the trained neural network to each patch in turn to produce a denoised output image. If the window size is too small, it may not capture enough information to properly denoise the image. If the window size is too large, it may include too much information, leading to overfitting, increased computation time, and reduced performance. The default window size of 5 is optimal for most images. 
+
+### Masking Ratio:
+
+The "masking ratio" parameter determines the proportion of pixels within each patch that are randomly masked during N2V training. By randomly setting some pixels within each patch to zero, N2V forces the neural network to learn a more robust representation of the underlying image structure, rather than simply memorizing the specific noise patterns in the input image. The masking ratio controls the level of noise injection during training, with higher values leading to more aggressive masking and generally better performance on noisy datasets. If the masking ratio is too small, the model may not be able to learn the noise distribution effectively, which can lead to poor denoising performance.
```

Comment:
This is one way of maksing that is described in the paper. However, the authors concluded that sampling replacement pixels from a surrounding area instead of assigning a value of 0 to the maked pixels leads to higher denoising performance.

Reply:
I changed the description as well.